src/main/java/de/tum/in/www1/artemis/service/connectors/bitbucket/BitbucketService.java

```diff
@@ -256,7 +236,92 @@ public void createUser(String username, String password, String emailAddress, St
     }
 
     /**
-     * Adds an Bitbucket user to (multiple) Bitbucket groups
+     * Updates an user on Bitbucket
+     *
+     * @param username     The username of the user
+     * @param emailAddress The new email address
+     * @param displayName  The new display name
+     * @throws BitbucketException
+     */
+    public void updateUserDetails(String username, String emailAddress, String displayName) throws BitbucketException {
+        UriComponentsBuilder userDetailsBuilder = UriComponentsBuilder.fromHttpUrl(bitbucketServerUrl + "/rest/api/latest/admin/users");
+        Map<String, Object> userDetailsBody = new HashMap<>();
+        userDetailsBody.put("name", username);
+        userDetailsBody.put("email", emailAddress);
+        userDetailsBody.put("displayName", displayName);
+        HttpEntity<Map<String, Object>> userDetailsEntity = new HttpEntity<>(userDetailsBody, null);
+
+        log.debug("Updating Bitbucket user {} ({})", username, emailAddress);
+
+        try {
+            restTemplate.exchange(userDetailsBuilder.build().encode().toUri(), HttpMethod.PUT, userDetailsEntity, Void.class);
+        }
+        catch (HttpClientErrorException e) {
+            if (isUserNotFoundException(e)) {
+                log.warn("Bitbucket user " + username + " does not exist.");
```

Comment:
```suggestion
                log.warn("Bitbucket user {} does not exist.", username);
```
I think using the built-in formatting of the logger is nicer than manual string concatenation. Also in the other places where no exception is passed to the logger.